### PR TITLE
Log more information when MHSM provisioning fails

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

File: sdk/keyvault/test-resources-post.ps1 [hunk 1]
```diff
@@ -95,7 +95,13 @@ if (Test-Path $sdpath) {
     Remove-Item $sdPath -Force
 }
 
-Export-AzKeyVaultSecurityDomain -Name $hsmName -Quorum 2 -Certificates $wrappingFiles -OutputPath $sdPath
+Export-AzKeyVaultSecurityDomain -Name $hsmName -Quorum 2 -Certificates $wrappingFiles -OutputPath $sdPath -ErrorAction SilentlyContinue -Verbose
+if ( !$? ) {
+    Write-Host $Error[0].Exception
+    Write-Error $Error[0]
+
+    exit
```
Comment (benbp) [inline sdk/keyvault/test-resources-post.ps1:103]: I assume this will still fail because of stderr for devops, right?

**Review by benbp — APPROVED** [review]

File: sdk/keyvault/test-resources-post.ps1 (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/test-resources-post.ps1:103]: Not sure. Didn't see a reason to take the risk. It's temporary anyway.

**Comment by heaths** [comment]
Appears to be a transient service issue, but the deployment passed for Managed HSM which is what we were trying to reproduce so I'm going to force this in. We'll leave the logging code in for now in case the issue with security domain download the last couple nights happens again.

**Pull request merged**